addons/calendar/models/calendar_event.py

```diff
@@ -695,7 +695,8 @@ def create(self, vals_list):
                 res_id = values.get('res_id', defaults.get('res_id'))
                 user_id = values.get('user_id', defaults.get('user_id'))
                 if not defaults.get('activity_ids') and res_model_id and res_id:
-                    if hasattr(self.env[self.env['ir.model'].sudo().browse(res_model_id).model], 'activity_ids'):
+                    res_model_name = self.env['ir.model'].sudo().browse(res_model_id).model
+                    if res_model_name in self.env and hasattr(self.env[res_model_name], 'activity_ids'):
```

Comment:
How are you getting a situation where your model_name is not in self.env ? Are you trying to create activities for non existent models or models not installed? You should explain what you are trying to solve in your  commit message.

Reply:
@arj-odoo Both, I use rpc to create activities for non existent models.
And for project migration from 13.0 to 14.0, when upgrading the module `calendar` before the module `project`, there will be an error because the module `calendar` doesn't depend on the module `project`, so models not installed (is not in self.env)